natlas-server/tests/elastic/test_interface.py

```diff
@@ -0,0 +1,13 @@
+from time import sleep
+
+from config import Config
+from app.elastic import ElasticInterface
+
+
+def test_delete_by_query():
+    client = ElasticInterface(Config().ELASTICSEARCH_URL)
+    client.new_result({
+        "ip": "127.0.0.1"
+    })
+    sleep(2)
+    assert client.delete_host("127.0.0.1") == 2
```

Comment:
Another potential test for this is calling `new_result()` twice and then confirming that `delete_host("127.0.0.1")` returns 3 results instead of 2. (Two documents in the `nmap_history` index, one document in the `nmap` index). 

The sleep appears arbitrary, is `new_result` asynchronously firing the request to elastic?

Reply:
The sleep is because Elasticsearch doesn't have immediate read-after-write by default. i.e It'll guarantee quorum write, but that doesn't make it available to reads upon acceptance of the write. I didn't dig into this further since the consistency may actually be immediate on a single node.